dapp/src/routes/DiscordVerification/DiscordVerification.svelte

```diff
@@ -0,0 +1,229 @@
+      lock = true;
+      func()
+        .then((res: any) => {
+          currentStep += 1;
+          resolve(res);
+        })
+        .catch(console.error)
+        .finally(() => (lock = false));
+    });
+  };
+</script>
+
+<BasePage
+  class="flex flex-grow text-white 2xl:px-32 px-8 overflow-hidden-x flex-wrap items-center justify-center fade-in"
+>
+  <div class="flex flex-col justify-evenly md:w-1/2">
+    <div
+      class="flex flex-col mb-4 transition-all ease-in-out duration-500 bg-white p-10 rounded-lg dropshadow-default"
+    >
+      <div
+        class="mb-4 text-2xl text-left font-bold body flex flex-row items-center"
+      >
+        <div class="mr-3">Discord Verification</div>
+      </div>
+      <div class="body">
+        This process is used to link your Discord account to your Tezos account
+        by signing a message using your private key, entering your Discord
+        username, and posting that message.
+      </div>
+    </div>
+
+    <VerificationStep
+      step={1}
+      bind:currentStep
+      title="Enter Discord Handle"
+      description="Enter your Discord account handle (ex: kevin#1234) to verify and include in a message signed via your wallet."
+    >
+      <div class="flex w-full mt-8">
+        <Input
+          placeholder="Enter your Discord handle"
+          class="mr-8"
+          bind:value={discordHandle}
+          disabled={currentStep !== 1}
+          name="enter-discord-handle"
+        />
+        {#if currentStep === 1}
+          <PrimaryButton
+            text="Submit"
+            onClick={() => {
+              next(() => getDiscordClaim($userData, discordHandle)).then(
+                (res) => {
+                  discordClaim = res;
+                  discordMessage = getDiscordMessage($userData, discordHandle);
+                }
+              );
+            }}
+            class="ml-4 lg:ml-0"
+            disabled={validateDiscordHandle(discordHandle)}
+            small
+          />
+        {/if}
+      </div>
+    </VerificationStep>
+
+    <VerificationStep
+      step={2}
+      bind:currentStep
+      title="Sign Prompt"
+      description="Sign the message presented to you containing your Discord handle and additional information."
+    >
+      {#if currentStep >= 2}
+        <div class="flex items-center w-full py-2 mt-8">
+          <textarea
+            class="overflow-x-auto rounded-lg bg-gray-100 body p-2 mr-4 w-full resize-none"
+            bind:value={discordClaim}
+            readonly
+            disabled
+          />
+          <CopyButton text={discordClaim} />
+        </div>
+      {/if}
+      {#if currentStep === 2}
+        <PrimaryButton
+          text="Signature Prompt"
+          class="mt-8 lg:w-48"
+          onClick={() => {
+            next(() =>
+              signDiscordClaim($userData, `${discordClaim}\n\n`, $wallet)
+            ).then(
+              (sig) =>
+                (discordMessage = `${getDiscordMessage(
```

Comment:
I think you can reuse `discordClaim` to avoid rebuilding the message.

Reply:
The two resulting messages are different so I think this will work.

Reply:
How could they be different? The message that's signed, and the message used in the verification need to be the same

Reply:
Refactored to new functions writteh by @krhoda  used to generate generalized social media verification messages.